libsnark/zk_proof_systems/plonk/prover.tcc

```diff
@@ -713,9 +716,12 @@ round_five_out_t<ppT> plonk_prover<ppT>::round_five(
 
     // --- Computation of r_part[3]
 
+    std::vector<polynomial<Field>> L_basis =
```

Comment:
Am I right that the Langrage basis is only ever generated in order to use the first element?  If so, why are we generating the entire thing?

If that first polynomial is required, it can  potentially go on the srs.  The idea was to try and not generate them if they are not required, but for things are definitely *are* required, we should mostly avoid generating them multiple times.

Reply:
Hmm, you make a very good point! 

> Am I right that the Langrage basis is only ever generated in order to use the first element?

Yes, it seems you are right and it is striking that this has escaped me. The only explanation that I have is that the paper is using the full basis e.g. in prover round 2 and I was doing so too in an initial version of the implementation.  Later I replaced the Lagrange basis computation by iFFT in most places and I didn't realize that in the remaining places we are only using the first polynomial from the basis.

Ok, so indeed let's have the first Lagrange polynomial in the srs then! In that case we won't need the domain in the srs anymore. Would you recommend to still keep it in case we may need it in the future (although premature optimization is generally not a good thing I guess)?

Reply:
Added the 0-th Lagrange basis polynomial in the SRS. Not computing the full basis on demand any more. The `domain` is still part of the SRS for now.